### PR TITLE
feat: Adding stencil.ReadDir to enumerate dir contents

### PR DESCRIPTION
**Pull request by deregtd**
Attempt to do basic validation/sanitization using billy for the dir contents you're reading

### CODE REVIEW

File: internal/codegen/tpl_stencil.go [hunk 1]
```diff
@@ -182,32 +181,73 @@ func (s *TplStencil) ReadFile(name string) (string, error) {
...
+	IsDir bool
+}
+
+// ReadDir reads the contents of a directory and returns a list of files/directories
+//
+//		{{ range $entry := stencil.ReadDir "/tests" }}
+//		  {{ if $entry.IsDir }}
+//		    {{ $entry.Name }}
+//	      {{ end }}
+//	    {{ end }}
```
Comment (jaredallard) [inline internal/codegen/tpl_stencil.go:195]: Indentation looks messed up here 🤔 (thx Go?)

I found I had to mess with tabs quite a bit to get this to work properly.

File: internal/codegen/tpl_stencil.go [hunk 2]
```diff
@@ -182,32 +181,73 @@ func (s *TplStencil) ReadFile(name string) (string, error) {
 	return string(b), nil
 }
 
+type ReadDirEntry struct {
+	Name  string
+	IsDir bool
+}
```
Comment (jaredallard) [inline internal/codegen/tpl_stencil.go:187]: **thought(non-blocking):** Is it worth just returning the [`os.FileInfo`](https://pkg.go.dev/os#FileInfo) interface instead?

File: internal/codegen/tpl_stencil_test.go [hunk 3]
```diff
@@ -322,3 +327,41 @@ func TestTplStencil_ApplyTemplate(t *testing.T) {
...
+
+	entries, err = s.ReadDir("testdata")
+	assert.NilError(t, err)
+	assert.Equal(t, true, len(entries) > 5)
+	assert.Equal(t, true, slices.ContainsFunc(entries, func(entry ReadDirEntry) bool {
+		return entry.Name == "blocks-test.txt" && !entry.IsDir
+	}))
+	assert.Equal(t, true, slices.ContainsFunc(entries, func(entry ReadDirEntry) bool {
+		return entry.Name == "args" && entry.IsDir
+	}))
```
Comment (jaredallard) [inline internal/codegen/tpl_stencil_test.go:366]: **thought:** This is kinda fragile if we add more files here ever

**Comment by jaredallard** [comment]
Looks like you're missing the docs generation step

File: internal/codegen/tpl_stencil.go (hunk 2 above)
Comment (deregtd) [reply internal/codegen/tpl_stencil.go:187]: can't -- that structure uses methods

File: internal/codegen/tpl_stencil_test.go (hunk 3 above)
Comment (deregtd) [reply internal/codegen/tpl_stencil_test.go:366]: i did > 5 just so it's got a stack and shouldn't need to touch it

File: internal/codegen/tpl_stencil.go (hunk 2 above)
Comment (jaredallard) [reply internal/codegen/tpl_stencil.go:187]: I'm pretty sure `file.Name` just gets called automatically in a Go template, or do you mean something else?

**Review by jaredallard — APPROVED** [review]

**Pull request merged**